### ISSUE TITLE
Issue in firefox when having big inline css

### ISSUE THREAD

**Issue by gabelerner** [canvg/canvg#171]
_From [edwin.zu...@gmail.com](https://code.google.com/u/108320155535737586973/) on September 12, 2012 04:43:58_

What steps will reproduce the problem? 1. Latest Firefox (15.0.1)
2. Try to render an svg to a canvas with an inline css of more then 255 lines
3. Only the first 255 lines are included and used for rendering What is the expected output? What do you see instead? This works fine in Chrome, all css should be included, but in Firefox everything past the 255th line is missing. What version of the product are you using? On what operating system? canvg1.2, Windows XP. Please provide any additional information below. This line in canvg.js (line 2189):

var css = node.childNodes[0].nodeValue + (node.childNodes.length > 1 ? node.childNodes[1].nodeValue : '');

Should be replaced with this:

var css = ''
for (var i=0; i<node.childNodes.length; i++) {
  css += node.childNodes[i].nodeValue;
}

Basically there is the wrong assumption that the whole css always fits into at most 2 childnodes.

_Original issue: http://code.google.com/p/canvg/issues/detail?id=171_
